src/LanguageServer/IdePurescript/Main.purs

```diff
@@ -466,16 +472,19 @@ handleEvents config conn state documents logError = do
 
   -- On document opened rebuild it,
   -- or place it in a queue if no IDE server started
-  onDidOpenDocument documents \{ document } -> launchAffLog do
-    mbPort <- liftEffect $ getPort state
-    case mbPort of
-      Just _ -> rebuildAndSendDiagnostics config conn state logError document
-      _ -> do
-        let uri = (un DocumentUri $ getUri document)
-        liftEffect $
-          Ref.modify_ (over ServerState (\st -> st
-          { buildQueue = Object.insert uri document (st.buildQueue)
-          })) state
+  onDidOpenDocument documents \{ document } ->
+    let uri = un DocumentUri $ getUri document
+    in case uri # isLibSourceFile of
```

Comment:
how about making handler a `do` block and 
```
let uri = un DocumentUri $ getUri document
when (isLibSourceFile uri) $ launchAffLog do ...
```

Reply:
Right, not used to use `when`, but it is absolutly appropriate here.

Reply:
Updated.